suite/tests/client-interface/drx-scattergather.templatex

```diff
@@ -1,29 +1,129 @@
 #ifdef __AVX512F__
+Expansion ok
+Expansion ok
+Expansion ok
 AVX-512 gather ok
+Expansion ok
+Expansion ok
+Expansion ok
 AVX-512 gather ok
+#ifdef X64
+Expansion ok
+Expansion ok
+Expansion ok
+#endif
 AVX-512 gather ok
+#ifdef X64
+Expansion ok
+Expansion ok
+Expansion ok
+#endif
 AVX-512 gather ok
+#ifdef X64
+Expansion ok
+Expansion ok
+Expansion ok
+#endif
 AVX-512 gather ok
+#ifdef X64
+Expansion ok
```

Comment:
suggest: Might be simpler to remove this print altogether and avoid all these ifdefs and complexity in this golden-output checking: rely instead on checks inside the test.

Reply:
I intentionally added that because should the expansion not occur, the test would pass. I actually ran into that when writing the patch and explicitly added the check. Leaving it as-is, re-open if necessary.

Reply:
Not following: whatever the test is doing to *print*, it can *assert* instead.  Generally, asserts are always preferable: easier to maintain; less flaky with little changes in output style; and easier to debug failures.  If something fails, an assert gives you a direct line number and direct callstack when attaching, instead of "missing 53rd printout" you have to go track back to the code.  This long series of identical prints looks like a nightmare for tracking down errors to me.  Travis will show the test output and the expected output and you'll stare at them wondering what the difference is; you'll have to paste into two files and diff them to find which line is off; then go spend a while mapping that back to the code.

Reply:
Ok, then. I pulled the check into an assertion. The only caveat is that I can't flat-rate check for expanded == true in 32-bit, because of the unsupported opcodes. I don't want to inspect the opcodes in the dll, I am checking only for 64-bit now. I think this should be ok, it's an additional extra check only anyway. Done.